serializer_impl.hh

```diff
@@ -782,4 +782,138 @@ unknown_variant_type deserialize(Input& in, boost::type<unknown_variant_type>) {
         return unknown_variant_type{ index, std::move(v) };
     });
 }
+
+template <typename T>
+class range_deserializer {
```

Comment:
Please add a comment explaining what this is about.

Reply:
sure